plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/launch/CLIParser.java

```diff
@@ -0,0 +1,108 @@
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.launch;
+
+import java.io.File;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.eclipse.glsp.server.utils.LaunchUtil;
+
+public final class CLIParser {
+   private static final Logger LOG = Logger.getLogger(CLIParser.class);
+   private final CommandLine cmd;
+   private final Options options;
+   private final String processName;
+
+   public CLIParser(final String[] args, final String processName) throws ParseException {
+      this(args, getDefaultCLIOptions(), processName);
+   }
+
+   public CLIParser(final String[] args, final Options options, final String processName) throws ParseException {
+      this.cmd = new DefaultParser().parse(options, args);
+      this.options = options;
+      this.processName = processName;
+   }
+
+   public boolean optionExists(final String identifier) {
+      return cmd.hasOption(identifier);
+   }
+
+   public Integer parsePort() {
+      String portArg = cmd.getOptionValue("p");
+      int port = LaunchUtil.DEFAULT_SERVER_PORT;
+      if (portArg != null) {
+         try {
+            port = Integer.parseInt(portArg);
+            if (!LaunchUtil.isValidPort(port)) {
+               throw new NumberFormatException();
+            }
+         } catch (NumberFormatException e) {
+            LOG.warn(String.format("'%s' is not a valid port! The default port '%s' is used",
+               portArg, LaunchUtil.DEFAULT_SERVER_PORT));
+         }
+      }
+
+      return port;
+   }
+
+   public String parseLogDir() {
+      String logDirArg = cmd.getOptionValue("d");
+      String logDir = LaunchUtil.DEFAULT_LOG_DIR;
+      if (logDirArg != null) {
+         File file = new File(logDirArg);
+         if (!file.exists() || !file.isDirectory()) {
+            LOG.warn(String.format("'%s' is not a valid directory! The default log path directory '%s' is used",
+               logDirArg, logDir));
+         } else {
+            logDir = file.getAbsolutePath();
+         }
+      }
+      return logDir;
+   }
+
+   public Level parseLogLevel() {
+      String levelArg = cmd.getOptionValue("ll");
+      return Level.toLevel(levelArg, LaunchUtil.DEFAULT_LOG_LEVEL);
+   }
+
+   public boolean isConsoleLog() { return optionExists("c"); }
+
+   public void printHelp() {
+      printHelp(this.processName, options);
+   }
+
+   public static void printHelp(final String processName, final Options options) {
```

Comment:
Super nit-picky: It seems to me that this method would be better located at the LaunchUtil class as it has no connection to the CLIParser whatsoever besides that it takes options.